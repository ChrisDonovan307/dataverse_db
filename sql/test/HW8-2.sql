-- 2. Select statements --------------------------------------------------

-- a
select e.Fname, e.Lname
from EMPLOYEE e
join EMPLOYEE s on e.Super_ssn = s.Ssn
where s.Fname = 'Franklin'
      and s.Lname = 'Wong';

-- b
select p.Pname, sum(w.Hours) as total_hours
from PROJECT p
join WORKS_ON w on p.Pnumber = w.Pno
group by p.Pname;

-- c
insert into WORKS_ON values (333445555, 1, NULL);
insert into WORKS_ON values (333445555, 30, NULL);

select e.Fname, e.Lname, p.Pname
from EMPLOYEE e
join WORKS_ON w on e.Ssn = w.Essn
join PROJECT p on w.Pno = p.Pnumber
order by p.Pname;

rollback;

-- d
delete from WORKS_ON
where Essn = 888665555
      and pNo = 20;

select e.Fname, e.Lname
from EMPLOYEE e
left join WORKS_ON w on e.Ssn = w.Essn
where w.Essn is null;

rollback;

-- e
select d.Dname, avg(e.Salary) as Avg_Salary
from EMPLOYEE e
join DEPARTMENT d on e.Dno = d.Dnumber
group by d.Dname;

-- f
select e.Fname, e.Lname, e.Address, l.Dnumber
from EMPLOYEE e
join WORKS_ON w on e.Ssn = w.Essn
join PROJECT p on w.Pno = p.Pnumber
join DEPT_LOCATIONS l on p.Dnum = l.Dnumber
join DEPARTMENT d on e.Dno = d.Dnumber
where p.Plocation = 'Houston'
      and l.Dnumber not in (
      	  select dep.Dnumber
	  from DEPARTMENT dep
	  join DEPT_LOCATIONS loc on dep.Dnumber = loc.Dnumber
	  where loc.Dlocation = 'Houston'
      );
