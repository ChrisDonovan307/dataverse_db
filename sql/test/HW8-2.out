SQL> 
SQL> -- Execute the script file. --
SQL> start &&1
SQL> -- 2. Select statements --------------------------------------------------
SQL> 
SQL> -- a
SQL> select e.Fname, e.Lname
  2  from EMPLOYEE e
  3  join EMPLOYEE s on e.Super_ssn = s.Ssn
  4  where s.Fname = 'Franklin'
  5  	   and s.Lname = 'Wong';

FNAME                                                                           
--------------------------------------------------                              
LNAME                                                                           
--------------------------------------------------                              
John                                                                            
Smith                                                                           
                                                                                
Ramesh                                                                          
Narayan                                                                         
                                                                                
Joyce                                                                           
English                                                                         
                                                                                

SQL> 
SQL> -- b
SQL> select p.Pname, sum(w.Hours) as total_hours
  2  from PROJECT p
  3  join WORKS_ON w on p.Pnumber = w.Pno
  4  group by p.Pname;

PNAME                     TOTAL_HOURS                                           
------------------------- -----------                                           
Newbenefits                        55                                           
ProductX                         52.5                                           
ProductY                         37.5                                           
Computerization                    55                                           
Reorganization                     25                                           
ProductZ                           50                                           

6 rows selected.

SQL> 
SQL> -- c
SQL> insert into WORKS_ON values (333445555, 1, NULL);

1 row created.

SQL> insert into WORKS_ON values (333445555, 30, NULL);

1 row created.

SQL> 
SQL> select e.Fname, e.Lname, p.Pname
  2  from EMPLOYEE e
  3  join WORKS_ON w on e.Ssn = w.Essn
  4  join PROJECT p on w.Pno = p.Pnumber
  5  order by p.Pname;

FNAME                                                                           
--------------------------------------------------                              
LNAME                                              PNAME                        
-------------------------------------------------- -------------------------    
Ahmad                                                                           
Jabbar                                             Computerization              
                                                                                
Alicia                                                                          
Zelaya                                             Computerization              
                                                                                
Franklin                                                                        
Wong                                               Computerization              
                                                                                

FNAME                                                                           
--------------------------------------------------                              
LNAME                                              PNAME                        
-------------------------------------------------- -------------------------    
Ahmad                                                                           
Jabbar                                             Newbenefits                  
                                                                                
Jennifer                                                                        
Wallace                                            Newbenefits                  
                                                                                
Alicia                                                                          
Zelaya                                             Newbenefits                  
                                                                                

FNAME                                                                           
--------------------------------------------------                              
LNAME                                              PNAME                        
-------------------------------------------------- -------------------------    
Franklin                                                                        
Wong                                               Newbenefits                  
                                                                                
John                                                                            
Smith                                              ProductX                     
                                                                                
Franklin                                                                        
Wong                                               ProductX                     
                                                                                

FNAME                                                                           
--------------------------------------------------                              
LNAME                                              PNAME                        
-------------------------------------------------- -------------------------    
Joyce                                                                           
English                                            ProductX                     
                                                                                
Joyce                                                                           
English                                            ProductY                     
                                                                                
John                                                                            
Smith                                              ProductY                     
                                                                                

FNAME                                                                           
--------------------------------------------------                              
LNAME                                              PNAME                        
-------------------------------------------------- -------------------------    
Franklin                                                                        
Wong                                               ProductY                     
                                                                                
Ramesh                                                                          
Narayan                                            ProductZ                     
                                                                                
Franklin                                                                        
Wong                                               ProductZ                     
                                                                                

FNAME                                                                           
--------------------------------------------------                              
LNAME                                              PNAME                        
-------------------------------------------------- -------------------------    
Jennifer                                                                        
Wallace                                            Reorganization               
                                                                                
Franklin                                                                        
Wong                                               Reorganization               
                                                                                
James                                                                           
Borg                                               Reorganization               
                                                                                

18 rows selected.

SQL> 
SQL> rollback;

Rollback complete.

SQL> 
SQL> -- d
SQL> delete from WORKS_ON
  2  where Essn = 888665555
  3  	   and pNo = 20;

1 row deleted.

SQL> 
SQL> select e.Fname, e.Lname
  2  from EMPLOYEE e
  3  left join WORKS_ON w on e.Ssn = w.Essn
  4  where w.Essn is null;

FNAME                                                                           
--------------------------------------------------                              
LNAME                                                                           
--------------------------------------------------                              
James                                                                           
Borg                                                                            
                                                                                

SQL> 
SQL> rollback;

Rollback complete.

SQL> 
SQL> -- e
SQL> select d.Dname, avg(e.Salary) as Avg_Salary
  2  from EMPLOYEE e
  3  join DEPARTMENT d on e.Dno = d.Dnumber
  4  group by d.Dname;

DNAME                     AVG_SALARY                                            
------------------------- ----------                                            
Headquarters                   55000                                            
Research                       33250                                            
Administration                 31000                                            

SQL> 
SQL> -- f
SQL> select e.Fname, e.Lname, e.Address, l.Dnumber
  2  from EMPLOYEE e
  3  join WORKS_ON w on e.Ssn = w.Essn
  4  join PROJECT p on w.Pno = p.Pnumber
  5  join DEPT_LOCATIONS l on p.Dnum = l.Dnumber
  6  join DEPARTMENT d on e.Dno = d.Dnumber
  7  where p.Plocation = 'Houston'
  8  	   and l.Dnumber not in (
  9  	       select dep.Dnumber
 10  	       from DEPARTMENT dep
 11  	       join DEPT_LOCATIONS loc on dep.Dnumber = loc.Dnumber
 12  	       where loc.Dlocation = 'Houston'
 13  	   );

no rows selected

SQL> SQL> 
SQL> -- Restore the options to their default values. --
SQL> spool off
